hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/fs/ozone/TestSnapshotDeletingService.java

```diff
@@ -0,0 +1,474 @@
+
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_ACL_ENABLED;
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_BLOCK_DELETING_SERVICE_INTERVAL;
+import static org.apache.hadoop.ozone.om.OmSnapshotManager.getSnapshotPrefix;
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_SNAPSHOT_DELETING_SERVICE_INTERVAL;
+import static org.apache.hadoop.ozone.OzoneConfigKeys.OZONE_SNAPSHOT_DELETING_SERVICE_TIMEOUT;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/**
+ * Test Snapshot Deleting Service.
+ */
+public class TestSnapshotDeletingService {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(TestSnapshotDeletingService.class);
+  private static boolean omRatisEnabled = true;
+  private static final String CONTENT = "testContent";
+
+  private MiniOzoneCluster cluster;
+  private OzoneManager om;
+  private OzoneBucket bucket1;
+  private OzoneClient client;
+  private static final String VOLUME_NAME = "vol1";
+  private static final String BUCKET_NAME_ONE = "bucket1";
+  private static final String BUCKET_NAME_TWO = "bucket2";
+
+  @BeforeEach
+  public void setup() throws Exception {
+    OzoneConfiguration conf = new OzoneConfiguration();
+    conf.setTimeDuration(OZONE_SNAPSHOT_DELETING_SERVICE_INTERVAL,
+        200, TimeUnit.MILLISECONDS);
+    conf.setTimeDuration(OZONE_SNAPSHOT_DELETING_SERVICE_TIMEOUT,
+        10000, TimeUnit.MILLISECONDS);
+    conf.setInt(OMConfigKeys.OZONE_DIR_DELETING_SERVICE_INTERVAL, 100);
+    conf.setInt(OMConfigKeys.OZONE_PATH_DELETING_LIMIT_PER_TASK, 5);
+    conf.setTimeDuration(OZONE_BLOCK_DELETING_SERVICE_INTERVAL, 100,
+        TimeUnit.MILLISECONDS);
+    conf.setBoolean(OMConfigKeys.OZONE_OM_RATIS_ENABLE_KEY, omRatisEnabled);
+    conf.setBoolean(OZONE_ACL_ENABLED, true);
+    // Enable filesystem snapshot feature for the test regardless of the default
+    conf.setBoolean(OMConfigKeys.OZONE_FILESYSTEM_SNAPSHOT_ENABLED_KEY, true);
+    cluster = MiniOzoneCluster.newBuilder(conf)
+        .setNumDatanodes(3)
+        .build();
+    cluster.waitForClusterToBeReady();
+    client = cluster.newClient();
+    om = cluster.getOzoneManager();
+    bucket1 = TestDataUtil.createVolumeAndBucket(
+        client, VOLUME_NAME, BUCKET_NAME_ONE, BucketLayout.DEFAULT);
+  }
+
+  @AfterEach
+  public void teardown() {
+    IOUtils.closeQuietly(client);
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  @Test
+  public void testSnapshotSplitAndMove() throws Exception {
+    SnapshotDeletingService snapshotDeletingService = (SnapshotDeletingService)
+        om.getKeyManager().getSnapshotDeletingService();
+    Table<String, SnapshotInfo> snapshotInfoTable =
+        om.getMetadataManager().getSnapshotInfoTable();
+
+    createSnapshotDataForBucket1();
+
+    assertTableRowCount(snapshotInfoTable, 2);
+    GenericTestUtils.waitFor(() -> snapshotDeletingService
+            .getSuccessfulRunCount() >= 1, 1000, 10000);
```

Comment:
nit: indentation. Can fix this in the other PR: https://github.com/apache/ozone/pull/4571/files#r1199182257

```suggestion
        .getSuccessfulRunCount() >= 1, 1000, 10000);
```